src/Models/RaspberryPiDevice.ts

```diff
@@ -74,118 +77,152 @@ export class RaspberryPiDevice implements Device {
-    const newPackageFilePath = path.join(deviceFolderPath, 'package.json');
+    return true;
+  }
 
-    try {
-      const packageObj = require(packageTemplateFilePath);
-      packageObj.main = sketchFileName;
-      fs.writeFileSync(newPackageFilePath, JSON.stringify(packageObj, null, 2));
-    } catch (error) {
-      throw new Error(`Create package.json failed: ${error.message}`);
-    }
+  async generateDockerRelatedFiles(): Promise<boolean> {
+        // Dockerfile       
+        const dockerfileTargetPath = path.join(
+          this.devcontainerFolderPath, FileNames.dockerfileName);
 
-    const settingsJSONFilePath =
-        path.join(vscodeFolderPath, FileNames.settingsJsonFileName);
-    const settingsJSONObj = {'files.exclude': {'.iotworkbenchproject': true}};
+        if (fs.existsSync(dockerfileTargetPath)) {
+          return true;
+        }
 
-    try {
-      fs.writeFileSync(
-          settingsJSONFilePath, JSON.stringify(settingsJSONObj, null, 4));
-    } catch (error) {
-      throw new Error(`Device: create config file failed: ${error.message}`);
-    }
+        try {
+          const dockerfileSourcePath = path.join(
+            this.boardFolderPath, RaspberryPiDevice.boardId, FileNames.dockerfileName);
+          const dockerfileContent = fs.readFileSync(dockerfileSourcePath, 'utf8');
+          fs.writeFileSync(dockerfileTargetPath, dockerfileContent);
+        } catch (error) {
+          throw new Error(`Create Dockerfile failed: ${error.message}`);
+        }
+    
+        // devcontainer.json
+        const devcontainerJSONFileTargetPath = path.join(
+          this.devcontainerFolderPath, FileNames.devcontainerJSONFileName);
 
-    cp.exec('npm install', {cwd: deviceFolderPath});
+        if (fs.existsSync(devcontainerJSONFileTargetPath)) {
+          return true;
+        }
+
+        try {
+          const devcontainerJSONFileSourcePath = path.join(
+            this.boardFolderPath, RaspberryPiDevice.boardId, FileNames.devcontainerJSONFileName);
+          const devcontainerJSONContent = fs.readFileSync(devcontainerJSONFileSourcePath, 'utf8');
+          fs.writeFileSync(devcontainerJSONFileTargetPath, devcontainerJSONContent);
+        } catch (error) {
+          throw new Error(`Create devcontainer.json file failed: ${error.message}`);
+        }
+
+        return true;
+  }
+  
+  async generateSketchFile(templateFilesInfo: TemplateFileInfo[]): Promise<boolean> {
+    // Generate sketch file
+    if (!templateFilesInfo) {
+      throw new Error('No sketch file found.');
+    }    
+    templateFilesInfo.forEach(fileInfo => {
+      const targetFilePath = path.join(
+            this.projectFolder, fileInfo.targetPath, fileInfo.fileName);
+      if (fileInfo.fileContent) {
+        try {
+          fs.writeFileSync(targetFilePath, fileInfo.fileContent);
+        } catch (error) {
+          throw new Error(
+              `Create Raspberrypi sketch file failed: ${error.message}`);

```

Comment:
Failed to create sketch file for Raspberry Pi: ${error.message}

Reply:
Are there any suggested error message style to refer to? That can can ensure consistensy.